python/ray/train/_internal/storage.py

```diff
@@ -98,7 +104,7 @@ def find(self, path, maxdepth=None, withdirs=False, detail=False, **kwargs):
 def _pyarrow_fs_copy_files(
     source, destination, source_filesystem=None, destination_filesystem=None, **kwargs
 ):
-    if isinstance(destination_filesystem, pyarrow.fs.S3FileSystem):
+    if S3FileSystem and isinstance(destination_filesystem, pyarrow.fs.S3FileSystem):
```

Comment:
I believe this needs to be `S3FileSystem` instead of `pyarrow.fs.S3FileSystem` or else the error will still be raised. I tried it locally by modifying pyarrow so I might be wrong:
```
In [3]: try:
   ...:     from pyarrow.fs import S3FileSystem
   ...: except ImportError:
   ...:     S3FileSystem = None
   ...: 

In [4]: pyarrow.fs.S3FileSystem
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[4], line 1
----> 1 pyarrow.fs.S3FileSystem

File ~/miniconda3/lib/python3.11/site-packages/pyarrow/fs.py:76, in __getattr__(name)
     74 def __getattr__(name):
     75     if name in _not_imported:
---> 76         raise ImportError(
     77             "The pyarrow installation is not built with support for "
     78             "'{0}'".format(name)
     79         )
     81     raise AttributeError(
     82         "module 'pyarrow.fs' has no attribute '{0}'".format(name)
     83     )

ImportError: The pyarrow installation is not built with support for 'S3FileSystem'
```